### PR TITLE
added assoicated class

### PR DESCRIPTION
**Pull request by Melaku05**
**Added class Classroom**

   Added  @label instance variable, and  initialized in the constructor.
    Added Setter and getter for @label (remember about attr_accessor).

Created  the has-many/belongs-to relationship between Classroom and Student, and  implemented:

    Created the has-many side (a classroom has many students).
    Created the belongs-to side (a student belongs to a classroom).

**Added  a class Book :**

    Added @title and @author instance variables, and initialized in the constructor.
    Added Setters and getters for instance variables (remember about attr_accessor).

**Added a class Rental:**

    Added @date instance variable, and initialized in the constructor.
    Added Setter and getter for @date (remember about attr_accessor).

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+  end
```
Comment (EricMbouwe) [inline classroom.rb]: - [x] When adding a student to a classroom you should also set the classroom for the student. ``student.classroom = self``

File: student.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class Student < Person
   def initialize(classroom, age, name = 'unknown', parent_permission: true)
     super(name, age, parent_permission)
-    @classroom = classroom
+    classroom.add_student(self) unless classroom.includes?(self)
```
Comment (EricMbouwe) [inline student.rb:8]: - [x] You need a separate method (eg: add_classroom) for this action. When setting the classroom for a student you should also add it to the classrooms' students.

File: person.rb [hunk 3]
```diff
@@ -10,6 +10,11 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
   end
```
Comment (EricMbouwe) [inline person.rb]: - [x] Adding rentals should be done by using the Renal class and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. Let's fix that

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi STUDENT, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Good PR
- No linters errors
- Commits
## Required Changes ♻️
- [ ] The `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` should be done by using the intermediate class `Rental` which you haven't implemented yet. Let's fix that
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (Melaku05) [reply classroom.rb]: Thank you, I did that

File: student.rb (hunk 2 above)
Comment (Melaku05) [reply student.rb:8]: Thank you, I did that

File: person.rb (hunk 3 above)
Comment (Melaku05) [reply person.rb]: Thank you, I did that

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 🟢  ✔</h1>
Hi @Melaku05, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**